samples/azure-quantum/entanglement-swap/ES-quantinuum-qiskit.ipynb

```diff
@@ -0,0 +1,930 @@
+       "      │  create_bell_state │ ░ │       │┌───┐┌───┐ ░ ┌───────┐           ░                     ░  ║    \n",
+       "qc_3: ┤1                   ├─░─┤       ├┤ X ├┤ Z ├─░─┤1      ├───────────░─────────────────────░──╫────\n",
+       "      ├────────────────────┤ ░ │       │└─╥─┘└─╥─┘ ░ │       │           ░                     ░  ║    \n",
+       "qc_4: ┤0                   ├─░─┤       ├──╫────╫───░─┤0      ├───────────░─────────────────────░──╫────\n",
+       "      │  create_bell_state │ ░ │       │  ║    ║   ░ │       │┌───┐┌───┐ ░ ┌───────┐           ░  ║    \n",
+       "qc_5: ┤1                   ├─░─┤  send ├──╫────╫───░─┤       ├┤ X ├┤ Z ├─░─┤1      ├───────────░──╫────\n",
+       "      ├────────────────────┤ ░ │       │  ║    ║   ░ │       │└─╥─┘└─╥─┘ ░ │       │           ░  ║    \n",
+       "qc_6: ┤0                   ├─░─┤       ├──╫────╫───░─┤  send ├──╫────╫───░─┤0      ├───────────░──╫────\n",
+       "      │  create_bell_state │ ░ │       │  ║    ║   ░ │       │  ║    ║   ░ │       │┌───┐┌───┐ ░  ║ ┌─┐\n",
+       "qc_7: ┤1                   ├─░─┤       ├──╫────╫───░─┤       ├──╫────╫───░─┤  send ├┤ X ├┤ Z ├─░──╫─┤M├\n",
+       "      └────────────────────┘ ░ │       │  ║    ║   ░ │       │  ║    ║   ░ │       │└─╥─┘└─╥─┘ ░  ║ └╥┘\n",
+       " cr1: ═════════════════════════╡0      ╞══■════╬═════╡0      ╞══■════╬═════╡0      ╞══■════╬══════╩══╬═\n",
+       "                               │       │ 0x1   ║     │       │ 0x1   ║     │       │ 0x1   ║         ║ \n",
+       " cr2: ═════════════════════════╡1      ╞═══════■═════╡1      ╞═══════■═════╡1      ╞═══════■═════════╩═\n",
+       "                               └───────┘      0x1    └───────┘      0x1    └───────┘      0x1          "
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def entanglement_swap_n(nParties: int): \n",
+    "    cr_x = ClassicalRegister(1, 'cr1')\n",
+    "    cr_z = ClassicalRegister(1, 'cr2')\n",
+    "    qc = QuantumCircuit(QuantumRegister(2 * nParties, 'qc'), cr_x, cr_z)\n",
+    "    for i in range(nParties):\n",
+    "        qc.append(create_bell_state(), [2 * i, 2 * i + 1])\n",
+    "    qc.barrier()\n",
+    "    for i in range(1, nParties):\n",
+    "        perform_teleportation_on_bell_state(qc, 2 * i, 2 * i + 1, 2 * i - 1, cr_x, cr_z)\n",
+    "        qc.barrier()\n",
+    "    # Measure bell state in classical registers\n",
+    "    qc.measure(0, cr_x)\n",
+    "    qc.measure(2 * nParties - 1, cr_z)\n",
+    "    return qc\n",
+    "\n",
+    "entanglement_swap_n(4).draw(fold=-1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We shall now test to see how fidelity changes with number of qubits. As our circuit includes mid-circuit measurements we will be running on Quantinuum hardware (/simulators) as they are the only current provider allowing such operations.\n",
```

Comment:
I'd strongly suggest avoiding the word "fidelity" unless you're doing a full state, process, or gateset tomography experiment and have been careful to run a robust statistical inference routine on the result. More generally, it may be good to avoid direct quantitative comparisons across different runs.